pico/runtime/src/main/java/io/helidon/pico/runtime/DefaultServices.java

```diff
@@ -216,46 +216,48 @@ public boolean reset(boolean deep) {
     }
 
     @Override
+    @SuppressWarnings({"unchecked", "rawtypes"})
     public <T> Optional<ServiceProvider<T>> lookupFirst(Class<T> type,
                                                         boolean expected) {
         DefaultServiceInfoCriteria criteria = DefaultServiceInfoCriteria.builder()
                 .addContractImplemented(type.getName())
                 .build();
-        return lookupFirst(criteria, expected);
+        return (Optional) lookupFirst(criteria, expected);
```

Comment:
This is exactly what I wanted to prevent. 
Is there a chance to add type to `ServiceInfoCriteria` instead? It could not use the generated builder, but it would be more user friendly

Reply:
No, someone can just lookup RunLevel.STARTUP services for example.  Type is not required to be passed.

Reply:
This change "fixes" the public contract at least, and the unfortunate casting is pushed down into the bowels of the impl.

Reply:
The default can be `ServiceInfoCriteria.Builder<Object>`, with a possibility to define a type (such as `<T> ServiceInfoCriteriaBuilder<T> contract(Class<T> type)`
This would allow us to make even internal implementation typed, while supporting use cases such as you describe

Reply:
It would make it more ambiguous. First, you would need to then choose to put the `T` on the `serviceTypeName`, `contractImplemented`, or `externalContractImplemented`. Also note that these do not use `Class` today - they use String.  This seems like a real bad idea to me - it will lead to confusion.

Reply:
`Contract` is not a singleton, it is a collection. Also, we would have to make analogous changes on `ServiceInfo` and possibly more builder types. It doesn't make sense.

Reply:
I think this would be more user friendly if we found a way, let's see once we use it ourself.